File: kotlin/src/main/kotlin/fizzbuzz/FizzBuzz.kt
package fizzbuzz

object FizzBuzz {

    private const val FIZZ_DIVISOR = 3
    private const val BUZZ_DIVISOR = 5
    private const val FIZZ = "Fizz"
    private const val BUZZ = "Buzz"

    fun of(n: Int): String {
        val result = buildString {
            if (n % FIZZ_DIVISOR == 0) append(FIZZ)
            if (n % BUZZ_DIVISOR == 0) append(BUZZ)
            if (isEmpty()) append(n)
        }
        return result
    }
}

File: node/src/fizzbuzz.js
const FIZZ_DIVISOR = 3;
const BUZZ_DIVISOR = 5;
const FIZZ = "Fizz";
const BUZZ = "Buzz";

export function of(n) {
  let result = "";
  if (n % FIZZ_DIVISOR === 0) result += FIZZ;
  if (n % BUZZ_DIVISOR === 0) result += BUZZ;
  if (result === "") result += n;
  return result;
}
